Sharpmake.Generators/VisualStudio/ProjectOptionsGenerator.cs

```diff
@@ -695,6 +695,16 @@ private void GenerateCompilerOptions(IGenerationContext context, ProjectOptionsG
             Options.Option(Options.Vc.Compiler.KeepComment.Enable, () => { context.Options["KeepComments"] = "true"; context.CommandLineOptions["KeepComments"] = "/C"; })
             );
 
+            //Options.Vc.Compiler.UseStandardConformingPreprocessor.    See: https://learn.microsoft.com/en-us/cpp/build/reference/zc-preprocessor?view=msvc-170
+            //    Disable                                 /Zc:preprocessor-
+            //    Enable                                  /Zc:preprocessor
+            context.SelectOption
+            (
+            Options.Option(Options.Vc.Compiler.UseStandardConformingPreprocessor.Default, () => { }),
+            Options.Option(Options.Vc.Compiler.UseStandardConformingPreprocessor.Disable, () => { context.Configuration.AdditionalCompilerOptions.Add("/Zc:preprocessor-"); }),
```

Comment:
this is wrong. Could can't modify Configuration here.

Reply:
The implementation should be similar to other options. See 
Options.Vc.Compiler.GenerateProcessorFile.Disable just above this code which is a very similar option.

Reply:
I added similar setup to Options.Vc.Compiler.GenerateProcessorFile.